go/libraries/doltcore/sqle/dsess/transactions.go

```diff
@@ -143,13 +143,55 @@ func doltCommit(ctx *sql.Context,
 	workingSet *doltdb.WorkingSet,
 	currHash hash.Hash,
 ) (*doltdb.WorkingSet, *doltdb.Commit, error) {
+	pending := *commit
+
 	headRef, err := workingSet.Ref().ToHeadRef()
 	if err != nil {
 		return nil, nil, err
 	}
 
+	headSpec, _ := doltdb.NewCommitSpec("HEAD")
+	curHead, err := tx.dbData.Ddb.Resolve(ctx, headSpec, headRef)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if curHead != nil {
+		curRootVal, err := curHead.ResolveRootValue(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+		curRootValHash, err := curRootVal.HashOf()
+		if err != nil {
+			return nil, nil, err
+		}
+		headRootValHash, err := pending.Roots.Head.HashOf()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if curRootValHash != headRootValHash {
+			start := time.Now()
+			mo := merge.MergeOpts{IsCherryPick: false}
+			pending.Roots.Staged, _, err = merge.MergeRoots(
```

Comment:
It bothers me that we're doing two merge operations here, one for the working set and one for the HEAD. It seems like it should be possible to unify the logic here and do a single merge to get our new root value, then set the root values in the pending commit as necessary.  Two merges is harder to reason about and is also going to be slower.

Am I missing something obvious about this? Isn't it the case that the working set and the HEAD will contain the same value after a dolt_commit operation? If so, why do the work to calculate that single value more than once?